js/util/string_number_mapping.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+module.exports = StringNumberMapping;
+
+function StringNumberMapping(strings) {
+    this.stringToNumber = {};
+    this.numberToString = [];
+    for (var i = 0; i < strings.length; i++) {
+        var string = strings[i];
+        this.stringToNumber[string] = i;
+        this.numberToString[i] = string;
+    }
+}
```

Comment:
From [wikipedia](https://en.wikipedia.org/wiki/Dictionary_coder), a dictionary coder is a 

> class of lossless data compression algorithms

Is there a noun for an instance of an algorithm?

`DictionaryCoder`, `DictionaryCoding`, `DictionaryCode`?
Or are they all the same thing?

`DictionaryCoder` sounds good enough to me.